src/Behat/Testwork/Output/Printer/JSONOutputPrinter.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * This file is part of the Behat Testwork.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Testwork\Output\Printer;
+
+use Behat\Testwork\Output\Exception\MissingOutputPathException;
+use Behat\Testwork\Output\Printer\Factory\FileOutputFactory;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class JSONOutputPrinter extends StreamOutputPrinter
```

Comment:
This is the class that generates the actual JSON, being called at the appropriate times by the different printers. It does this by generating an array of all the nodes and then converting this array to json before saving it to a file